### PR TITLE
Make the GUI great again!

### PR DESCRIPTION
**Pull request by katekerllenevich**
I can't actually test this, my system segfaults whenever I try to load the GUI, if anyone wants to test it, feel free.

### CODE REVIEW

File: patches/server/0259-Make-the-GUI-Better.patch [hunk 1]
```diff
@@ -0,0 +1,1088 @@
...
+diff --git a/src/main/java/net/pl3x/purpur/gui/console/JColorTextPane.java b/src/main/java/net/pl3x/purpur/gui/console/JColorTextPane.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..cfbcfdb73e4aeb2244ae8d22db3dc67869b47dde
+--- /dev/null
++++ b/src/main/java/net/pl3x/purpur/gui/console/JColorTextPane.java
+@@ -0,0 +1,79 @@
++package net.pl3x.purpur.gui.console;
++
++import com.google.common.collect.Sets;
++import net.md_5.bungee.api.ChatColor;
```
Comment (katekerllenevich) [inline patches/server/0259-Make-the-GUI-Better.patch]: Needs more adventure

File: patches/server/0259-Make-the-GUI-Better.patch [hunk 2]
```diff
@@ -0,0 +1,1088 @@
...
++}
+diff --git a/src/main/java/net/pl3x/purpur/gui/util/GUIColor.java b/src/main/java/net/pl3x/purpur/gui/util/GUIColor.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..973c8ddf93578b99079c036a305b0199eed2a0ad
+--- /dev/null
++++ b/src/main/java/net/pl3x/purpur/gui/util/GUIColor.java
+@@ -0,0 +1,54 @@
++package net.pl3x.purpur.gui.util;
++
++import net.md_5.bungee.api.ChatColor;
```
Comment (katekerllenevich) [inline patches/server/0259-Make-the-GUI-Better.patch]: Needs more adventure

**Comment by CodeDoctorDE** [comment]
Any updates?

**Comment by katekerllenevich** [comment]
Nope

**Comment by SageSphinx63920** [comment]
I tried to build it but I had 20 errors ans I just doesn't start .-.

**Comment by SageSphinx63920** [comment]
Could you force push it to the news 1.18.2 branch?

**Comment by katekerllenevich** [comment]
So my hand is actually in a brace right now so I really couldn't

**Comment by SageSphinx63920** [comment]
@BillyGalbreath we need your help

**Comment by granny** [comment]
It's not high priority. If it happens it happens.

**Comment by SageSphinx63920** [comment]
Yeah! Maybe I can finally build this myself now

**Comment by granny** [comment]
Currently in the process of testing it out. If it works with no issues, I'm merging it back in.

**Comment by granny** [comment]
So the main two features this PR aims to add (text formatting & command history) don't actually work. 

TODO:
- [x] fix command history
- [ ] fix text formatting
- [ ] add tab completion

**Pull request closed without merging**